src/ipc/Queue.h

```diff
@@ -410,6 +426,96 @@ OneToOneUniQueue::push(const Value &value, QueueReader *const reader)
     return wasEmpty && (!reader || reader->raiseSignal());
 }
 
+template <class Value>
+void
+OneToOneUniQueue::statIn(StoreEntry &entry) const
+{
+    // Nobody can modify our theIn so, after capturing some valid theSize value
+    // in count, we can reliably report all [theIn-count, theIn) items that were
+    // queued at theSize capturing time. We may report items already pop()ed by
+    // the other side, but that is OK because pop() does not modify items -- it
+    // only increments theOut.
+    const auto count = theSize.load();
+    statOpen(entry, "pushIndex", "other", count);
+    statClose<Value>(entry, theIn - count, count); // Branch XXX: theIn-count may get negative!
+}
+
+template <class Value>
+void
+OneToOneUniQueue::statOut(StoreEntry &entry) const
+{
+    // Nobody can modify our theOut so, after capturing some valid theSize value
+    // in count, we can reliably report all [theOut, theOut+count) items that
+    // were queued at theSize capturing time. We will miss new items push()ed by
+    // the other side, but that is OK.
```

Comment:
The same suggestion: 'theOut' index is modified by the other side pop() - we need to move this comment into statIn().

Reply:
See https://github.com/measurement-factory/squid/pull/96#discussion_r448326221